plugins/inputs/aos/aos.go

```diff
@@ -336,261 +402,351 @@ func (ssl *StreamAos) MsgReader(r io.Reader) {
 			return
 		}
 
-		// Create new aos_streaming.AosMessage and deserialize protobuf
-		newMsg := new(aos_streaming.AosMessage)
-		err = proto.Unmarshal(msgBuf, newMsg)
+		ssl.handleTelemetry(msgBuf)
+	}
+}
+
+func (ssl *StreamAos) handleTelemetry(msgBuf []byte) {
+	// Create new aos_streaming.AosMessage and deserialize protobuf
+	newMsg := new(aos_streaming.AosMessage)
+	sequencedMsg := new(aos_streaming.AosSequencedMessage)
+
+	ssl.sequencingMode.Do(func() {
+		// Automatically detect if we are receiving sequenced or unsequenced protobuf messages
+		err := proto.Unmarshal(msgBuf, sequencedMsg)
+		if err != nil {
+			log.Printf("W! Error unmarshaling sequenced message: %v", err)
+			return
+		} else {
+			err = proto.Unmarshal(sequencedMsg.GetAosProto(), newMsg)
+			if err != nil {
+				log.Printf("W! Message seq_num not supported: %v trying unsequenced", err)
+				err = proto.Unmarshal(msgBuf, newMsg)
+				if err != nil {
+					log.Printf("W! Error unmarshaling: %v", err)
+					return
+				}
+				ssl.IsSequencedStream = false
+			} else {
+				ssl.IsSequencedStream = true
+			}
+		}
+	})
 
+	if ssl.IsSequencedStream {
+		err := proto.Unmarshal(msgBuf, sequencedMsg)
+		if err != nil {
+			log.Printf("W! Error unmarshaling sequenced message: %v", err)
+			return
+		} else {
+			err = proto.Unmarshal(sequencedMsg.GetAosProto(), newMsg)
+			if err != nil {
+				log.Printf("W! Error unmarshaling sequenced aos_proto message: %v", err)
+			}
+		}
+	} else {
+		// Handle the original unsequenced message format
+		err := proto.Unmarshal(msgBuf, newMsg)
 		if err != nil {
 			log.Printf("W! Error unmarshaling: %v", err)
 			return
 		}
+	}
+
+	// ----------------------------------------------------------------
+	// Extract all Types of data
+	// ----------------------------------------------------------------
+	newPerfMonData := newMsg.GetPerfMon()
```

Comment:
Suggest consistent naming. "data" suffix is not there for other two variables - newEvent, newAlert

Reply:
Changed suffix.